### DISCUSSION TITLE
Email Provider redirect to callbackURL after verification not working

### DISCUSSION THREAD

**Discussion by bradbarrow** [nextauthjs/next-auth#2585] [Help]
Hi folks, thanks in advance for any assistance.

## Context

`next-auth@3.23.2`
`next@10.2.1`
Email Provider

## Steps

User visits `/landing-page`
User types their email address and submits the form
We call `signIn`:

```
signIn('email', {
  email: 'email@address.com',
  redirect: false, // stay on the page, show a message in-place
})
```

We don't redirect, we show a message saying "Check your email to verify"

## The Issue

User follows the link in their email
User gets created, session gets populated
User is redirected to the home page **when I expected them to be taken to /landing-page**

## What I have tried

I've tried adding an explicit `callbackUrl` to the `signIn` call
I've checked all the values of `NEXTAUTH_URL` etc

-----

Is there any way to achieve this? Redirecting the user back to the page that initiated a non-redirect `signIn` when they follow the email verification

**Comment by balazsorban44** [comment]
In case it is a new user, check the `pages.newUser` option: https://next-auth.js.org/configuration/pages
Otherwise here is a possible related discussion: #2574
